### ISSUE TITLE
Pygame error after window resize when using AsyncImage

### ISSUE THREAD

**Issue by voen** [kivy/kivy#2103]
Run the code below and resize the window after the image is loaded, the image should disappear. Tested on windows 8.1, python 3.4, pygame 1.9.2.
```python
from kivy.app import App
from kivy.lang import Builder


kv = """
AnchorLayout:
    AsyncImage:
        source: 'https://i.imgur.com/OS2E1.jpg'
        size_hint: None, None
        size: 500, 500
"""


class TestApp(App):
    def build(self):
        return Builder.load_string(kv)

if __name__ == '__main__':
    TestApp().run()
```

Console log:
```python
[INFO              ] [Logger      ] Record log in C:\Users\user\.kivy\logs\kivy_14-04-22_18.txt
[INFO              ] Kivy v1.8.1-dev
[INFO              ] [Python      ] v3.4.0 (v3.4.0:04f714765c13, Mar 16 2014, 19:24:06) [MSC v.1600 32 bit (Intel)]
[INFO              ] [Factory     ] 157 symbols loaded
[INFO              ] [Image       ] Providers: img_tex, img_dds, img_pygame, img_pil, img_gif 
[INFO              ] [Window      ] Provider: pygame(['window_egl_rpi'] ignored)
libpng warning: iCCP: known incorrect sRGB profile
GLEW initialization succeeded
[INFO              ] [GL          ] OpenGL version <b'3.3.0'>
[INFO              ] [GL          ] OpenGL vendor <b'NVIDIA Corporation'>
[INFO              ] [GL          ] OpenGL renderer <b'GeForce 9600 GSO/PCIe/SSE2'>
[INFO              ] [GL          ] OpenGL parsed version: 3, 3
[INFO              ] [GL          ] Shading version <b'3.30 NVIDIA via Cg compiler'>
[INFO              ] [GL          ] Texture max size <8192>
[INFO              ] [GL          ] Texture max units <32>
[INFO              ] [Window      ] virtual keyboard not allowed, single mode, not docked
[INFO              ] [Loader      ] using a thread pool of 2 workers
[INFO              ] [OSC         ] using <thread> for socket
[INFO              ] [Base        ] Start application main loop
[INFO              ] [GL          ] NPOT texture support is available
libpng warning: iCCP: known incorrect sRGB profile
[INFO              ] [Context     ] Reloading graphics data...
[WARNING           ] [Image       ] Unable to load image <None>
 Exception ignored in: 'kivy.graphics.texture.Texture.reload'
 Traceback (most recent call last):
   File "C:\dev\python\lib\site-packages\kivy\core\image\__init__.py", line 444, in __init__
     self.filename = arg
   File "C:\dev\python\lib\site-packages\kivy\core\image\__init__.py", line 633, in _set_filename
     mipmap=self._mipmap, nocache=self._nocache)
   File "C:\dev\python\lib\site-packages\kivy\core\image\__init__.py", line 371, in load
     im = loader(filename, **kwargs)
   File "C:\dev\python\lib\site-packages\kivy\core\image\__init__.py", line 150, in __init__
     self._data = self.load(filename)
   File "C:\dev\python\lib\site-packages\kivy\core\image\img_pygame.py", line 41, in load
     im = pygame.image.load(filename)
 pygame.error: Can't seek in this data source
[INFO              ] [Context     ] Reloading done in 0.0160s
```

**Comment by dessant** [comment]
This issue is fixed in 1.9.0 with SDL2, tested on win and debian.

**Comment by myishh** [comment]
Could tell me specifically how to fix the warning
